libstuff/STCPServer.cpp

```diff
@@ -23,6 +23,17 @@ STCPServer::Port* STCPServer::openPort(const string& host) {
     return &*portIt;
 }
 
+
+void STCPServer::shutdownPorts() {
```

Comment:
There's a description of the difference [here](https://stackoverflow.com/questions/4160347/close-vs-shutdown-socket). Basically, shutdown stops you receiving (or sending) data over the network, but lets that happen gracefully. You can still read from your receive buffer until it's empty. This lets us deserialize anything left in the buffer, and then we'll `close()` the connection when get to the end of `postPoll`.